src/lib/src/service/execution/test.execution.service.ts

```diff
@@ -17,9 +23,9 @@ export interface TestExecutionStatus {
 
 
 export abstract class TestExecutionService {
-  abstract execute(path: string): Promise<any>;
-  abstract getStatus(path: string): Promise<TestExecutionStatus>;
-  abstract getAllStatus(): Promise<TestExecutionStatus[]>;
+  abstract execute(path: string, onThen?: (some: any) => void, onError?: (error: any) => void): void
```

Comment:
just if its low-impact: consider renaming "onThen" to "onResponse" (just think the former sounds a bit weird…)